packages/wix-style-react/src/PulseAnimation/PulseAnimation.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { st, classes } from './PulseAnimation.st.css';
+import Animate from '../Animate';
+import { childSize, childWidthRange } from './constants';
+
+const isValueInRange = (x, min, max) => x >= min && x <= max;
+
+/** PulseAnimation*/
+class PulseAnimation extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      animationSize: undefined,
+    };
+    this.rootRef = React.createRef();
+  }
+
+  componentDidMount() {
+    const animationSize = this._getAnimationSize();
+    this.setState({ animationSize });
+  }
+
+  _getAnimationSize = () => {
+    const childWidth = this.rootRef.current.offsetWidth;
+    const { from, to } = childWidthRange.medium;
+
+    return isValueInRange(childWidth, from, to)
+      ? childSize.medium
+      : childSize.small;
+  };
+
+  render() {
+    const {
+      dataHook,
+      active,
+      onEnd,
+      onStart,
+      loop,
+      children,
+      delay,
+      color,
+      borderRadius,
+    } = this.props;
+
+    const { animationSize } = this.state;
+
+    return (
+      <Animate
+        dataHook={dataHook}
+        delay={delay}
+        className={st(classes.root, {
+          active,
+          loop,
+          size: animationSize,
+          color,
+          borderRadius,
+        })}
+        onEnd={onEnd}
+        onStart={onStart}
+        ref={this.rootRef}
+        borderRadius={borderRadius}
```

Comment:
Let's rename this prop to `animateInlineStyle`. Moreover, I think that using css variable for the inline styling is a better solution. 

```suggestion
        animateInlineStyle={[ [vars.borderRadius]: borderRadius ] }
```


There's an example with usage of [css variable](https://github.com/wix/wix-style-react/blob/master/packages/wix-style-react/src/CardFolderTabs/CardFolderTabs.js#L23)